pkg/arvo/sur/group.hoon

```diff
@@ -0,0 +1,94 @@
+^?
+|%
+::  $group-id: unique identifier for a group
+::
++$  group-id  [=ship =term]
+::  $groups: a mapping from group-ids to groups
+::
++$  groups  (map group-id group)
+::  $group-tag: an identifier used by groups
+::
+::    These tags should have precise semantics, as they are shared across all
+::    apps.
+::
++$  group-tag  ?(role-tag)
+::  $tag: an identifier used to identify a subset of members
+::
+::   Tags may be used and recognised differently across apps.
+::   for example, you could use tags like `%author`, `%bot`, `%flagged`...
+::
++$  tag  $@(group-tag [app=term tag=term])
+::  $role-tag: a kind of $group-tag that identifies a privileged user
+::
+::    These roles are
+::    %admin: Administrator, can do everything except delete the group
+::    %moderator: Moderator, can add/remove/ban users
+::    %janitor: Has no special meaning inside group-store,
+::    but may be given additional privileges in other apps.
+::
++$  role-tag
+  ?(%admin %moderator %janitor)
+::  $tags: a mapping from a $tag to the members it identifies
+::
++$  tags  (jug tag ship)
+::  $group: description of a group of users
+::
+::    .members: members of the group
+::    .tag-queries: a map of subsets
+::    .policy: permissions for the group
+::    .hidden: is group unmanaged
++$  group
+  $:  members=(set ship)
+      =tags
+      =policy
+      hidden=_&
+  ==
+::  $policy: access control for a group
+::
+++  policy
+  =<  policy
+  |%
+  ::
+  +$  policy
+    $%  invite
+        open
+    ==
+  ::  $diff: change group policy
+  +$  diff
+    $%  diff:invite
+        diff:open
```

Comment:
Maybe you want these to be tagged with the policy tag also, to avoid diff tag conflicts. (Not the case _right now_ but, you know, structure sanity.)

```suggestion
    $%  [%invite diff:invite]
        [%open diff:open]
```